model/span.go

```diff
@@ -77,15 +101,23 @@ func (s *Span) GetSpanKind() (spanKind trace.SpanKind, found bool) {
 }
 
 // GetSamplerType returns the sampler type for span
-func (s *Span) GetSamplerType() string {
+func (s *Span) GetSamplerType() SamplerType {
 	// There's no corresponding opentelemetry tag label corresponding to sampler.type
-	if tag, ok := KeyValues(s.Tags).FindByKey(samplerType); ok {
-		if tag.VStr == "" {
-			return samplerTypeUnknown
+	if tag, ok := KeyValues(s.Tags).FindByKey(keySamplerType); ok {
+		switch tag.VStr {
+		case SamplerTypeProbabilistic.String():
```

Comment:
this doesn't make sense to me. String() is already implemented as a switch statement, which is sequential comparison, so you're using O(n^2) resolution here, instead of O(1) when you had a map.

Reply:
> Why use map for static lookups? It's a waste of cpu cycles

Sorry! but you told map is not needed. Or maybe I misunderstood it

Reply:
you're taking it out of context. 
  * `someMap[someLiteralConstant]` is a waste of CPU. 
  * `someMap[someVariable]` is not a waste, but likely more efficient than a switch statement